tests/specs/unit/core/hello.login.js

```diff
@@ -154,6 +154,98 @@ describe('hello.login', function() {
 			hello.login('testable', {redirect_uri: REDIRECT_URI});
 		});
 
+		it('yahoo : should base 64 encode the state by default, if oauth option is not overridden', function(done) {
+
+			var spy = sinon.spy(function(url, name, optins) {
+				// The url should not contain uri encoded characters
+				expect(url).to.not.contain('state=%7B%22');
+
+				done();
+			});
+
+			utils.popup = spy;
+
+			hello.login('yahoo');
+		});
+
+		it('yahoo : should base 64 encode the state if base64_state is true', function(done) {
+
+			hello.services.yahoo.oauth.base64_state = true;
+
+			var spy = sinon.spy(function(url, name, optins) {
+				// The url should not contain uri encoded characters
+				expect(url).to.not.contain('state=%7B%22');
+
+				done();
+			});
+
+			utils.popup = spy;
+
+			hello.login('yahoo');
+		});
```

Comment:
We dont define individual module settings here...

```suggestion
```

But the other tests are great 👍